src/main/resources/static/js/main.js

```diff
@@ -150,47 +149,65 @@
         latitude: position.coords.latitude,
         longitude: position.coords.longitude
       });
+      geometry=true;
       locationButton.toggleClass('loading');
     }, function () {
       locationButton.toggleClass('loading');
       locationButton.toggleClass('disabled');
     })
   });
 
-  const geocodingButton = $('.ui.button#geocoding'); // it is not used
   const card = $('#pac-card');
   const input = document.getElementById('pac-input');
-  //const input = $('.ui.input#pac-input');
   const infowindowContent = document.getElementById('infowindow-content');
-  //const infowindowContent = $('#infowindow-content');
+  const politicalBoundary = 'political'; // Bee.place just contains Administrative/Political location
   const geoCoding = function() {
-    geometry=false;
     var autocomplete = new google.maps.places.Autocomplete(input);
     var infowindow = new google.maps.InfoWindow();
     infowindow.setContent(infowindowContent);
     autocomplete.addListener('place_changed', function() {
-    infowindow.close();
-    var place = autocomplete.getPlace();
-      if (place.geometry) {
+      infowindow.close();
+      geometry=false;
+      var place = autocomplete.getPlace();
+      if (place.geometry) { // User has selected a place from the autocomplete dropdown menu
         geometry=true;
+        var politicalPlaces = [];
         var place_output_str = '';
         if (place.address_components) {
-          //TODO place: just select political types: country - administrative_area (level 1 and 2). Not streets and numbers
-          place_output_str = [
-            (place.address_components[0] && place.address_components[0].long_name || ''),
-            (place.address_components[1] && place.address_components[1].long_name || ''),
-            (place.address_components[2] && place.address_components[2].long_name || '')
-            ].join(' ');
+          for (p in place.address_components) {
+            var politicalFlag = false;
+            for (t in place.address_components[p].types){
+               if (place.address_components[p].types[t] == politicalBoundary){
+                  politicalFlag = true;
+                  break;
+                }
+              }
+            if (politicalFlag && place.address_components[p]){
+                politicalPlaces.push(place.address_components[p].long_name)
+            }
+          }
+        }
+        // Bee.place formatting: deleting duplicates entries (City = Municipality, ie 'Calle Bami 14, Seville') and separated by comma
+        var uniquePoliticalPlaces = [];
+        $.each(politicalPlaces, function(i, el){
```

Comment:
let's keep `for .. in ..` syntax if possible.

Reply:
`for .. in ..` make things more complicated for this particular case.

As an alternative, we could use [JS filter|https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter] see this example http://mikeheavers.com/tutorials/removing_duplicates_in_an_array_using_javascript/